lib/bulk/common.js

```diff
@@ -427,26 +460,560 @@ function handleMongoWriteConcernError(batch, bulkResult, ordered, err, callback)
  * @return {BulkWriteError} A BulkWriteError instance
  * @extends {MongoError}
  */
-const BulkWriteError = function(error, result) {
-  var message = error.err || error.errmsg || error.errMessage || error;
-  MongoError.call(this, message);
+class BulkWriteError extends MongoError {
+  constructor(error, result) {
+    const message = error.err || error.errmsg || error.errMessage || error;
+    super(message);
 
-  var keys = typeof error === 'object' ? Object.keys(error) : [];
-  for (var i = 0; i < keys.length; i++) {
-    this[keys[i]] = error[keys[i]];
+    Object.assign(this, error);
+
+    this.name = 'BulkWriteError';
+    this.result = result;
   }
+}
 
-  this.name = 'BulkWriteError';
-  this.result = result;
-};
-util.inherits(BulkWriteError, MongoError);
+class FindOperators {
+  constructor(self, options) {
+    this.s = Object.assign(self.s, options);
+  }
+  /**
+   * Add a single update document to the bulk operation
+   *
+   * @method
+   * @param {object} updateDocument update operations
+   * @throws {MongoError}
+   * @return {OrderedBulkOperation or UnordedBulkOperation}
+   */
+  update(updateDocument) {
+    // Perform upsert
+    const upsert = typeof this.s.currentOp.upsert === 'boolean' ? this.s.currentOp.upsert : false;
+
+    // Establish the update command
+    const document = {
+      q: this.s.currentOp.selector,
+      u: updateDocument,
+      multi: true,
+      upsert: upsert
+    };
+
+    // Clear out current Op
+    this.s.currentOp = null;
+    return this.s.addToOperationsList(this, UPDATE, document);
+  }
+
+  /**
+   * Add a single update one document to the bulk operation
+   *
+   * @method
+   * @param {object} updateDocument update operations
+   * @throws {MongoError}
+   * @return {OrderedBulkOperation or UnordedBulkOperation}
+   */
+  updateOne(updateDocument) {
+    // Perform upsert
+    const upsert = typeof this.s.currentOp.upsert === 'boolean' ? this.s.currentOp.upsert : false;
+
+    // Establish the update command
+    const document = {
+      q: this.s.currentOp.selector,
+      u: updateDocument,
+      multi: false,
+      upsert: upsert
+    };
+
+    // Clear out current Op
+    this.s.currentOp = null;
+    return this.s.addToOperationsList(this, UPDATE, document);
+  }
+
+  /**
+   * Add a replace one operation to the bulk operation
+   *
+   * @method
+   * @param {object} updateDocument the new document to replace the existing one with
+   * @throws {MongoError}
+   * @return {OrderedBulkOperation or UnorderedBulkOperation}
```

Comment:
I think the actual format for this is: `@returns {(OrderedBulkOperation|UnorderedBulkOperation)}`, and there are a few more cases like this.